src/main/java/com/cosmian/jna/findex/Findex.java

```diff
@@ -178,12 +218,26 @@ public SearchRequest(byte[] key, byte[] label, Database database) {
             this.database = database;
         }
 
+        public SearchRequest(byte[] key, byte[] label, int entryTableNumber, Database database) {
+            this.key = key;
+            this.label = label;
+            this.database = database;
+            this.entryTableNumber = entryTableNumber;
+        }
+
```

Comment:
I don't think the constructor is necessary you could expose a setter instead since it's a rarely use feature.